Helper/Data.php

```diff
@@ -553,10 +553,6 @@ private function getCategoryRecords($storeId, $collection, $potentiallyDeletedCa
                 unset($potentiallyDeletedCategoriesIds[$categoryId]);
             }
 
-            if (isset($categorysToIndex[$categoryId]) || isset($categorysToRemove[$categoryId])) {
```

Comment:
I think there was a mistake here and the line should be 
`
if (isset($categoriesToIndex[$categoryId]) || isset($categoriesToRemove[$categoryId])) {`